File: sdks/php/src/Wallets/Requests/UpdateWalletDto.php
<?php

namespace NovaBilling\Wallets\Requests;

use NovaBilling\Core\Json\JsonSerializableType;
use NovaBilling\Core\Json\JsonProperty;
use NovaBilling\Core\Types\ArrayType;

class UpdateWalletDto extends JsonSerializableType
{
    /**
     * @var ?string $name
     */
    #[JsonProperty('name')]
    public ?string $name;

    /**
     * @var ?string $expirationAt
     */
    #[JsonProperty('expirationAt')]
    public ?string $expirationAt;

    /**
     * @var ?array<string, mixed> $metadata
     */
    #[JsonProperty('metadata'), ArrayType(['string' => 'mixed'])]
    public ?array $metadata;

    /**
     * @param array{
     *   name?: ?string,
     *   expirationAt?: ?string,
     *   metadata?: ?array<string, mixed>,
     * } $values
     */
    public function __construct(
        array $values = [],
    ) {
        $this->name = $values['name'] ?? null;
        $this->expirationAt = $values['expirationAt'] ?? null;
        $this->metadata = $values['metadata'] ?? null;
    }
}

File: sdks/ruby/lib/novabilling/wallets/types/update_wallet_dto.rb
# frozen_string_literal: true

module Novabilling
  module Wallets
    module Types
      class UpdateWalletDto < Internal::Types::Model
        field :id, -> { String }, optional: false, nullable: false
        field :name, -> { String }, optional: true, nullable: false
        field :expiration_at, -> { String }, optional: true, nullable: false, api_name: "expirationAt"
        field :metadata, -> { Internal::Types::Hash[String, Object] }, optional: true, nullable: false
      end
    end
  end
end
